services/payments/secrets.go

```diff
@@ -260,17 +386,32 @@ func (s *Service) decryptSecrets(ctx context.Context) (map[string]string, error)
 		return nil, fmt.Errorf("failed to parse private key bytes for secret decryption: %w", err)
 	}
 
-	buf := bytes.NewBuffer(s.secretsCiphertext)
+	return age.Decrypt(bytes.NewReader(buf.Bytes()), identity)
+}
 
-	r, err := age.Decrypt(buf, identity)
+func (s *Service) encryptWithShares(ctx context.Context, data []byte) (io.Reader, error) {
+	// combine the service configured key shares
+	privateKey, err := shamir.Combine(s.keyShares)
```

Comment:
now that the keyshares have been combined, can we overwrite them to not have them lingering in memory beyond when necessary? We may not be able to do this one because we want the keyShares to be available to decrypt the config at potentially different times, but I'm not certain yet so wanted to start a discussion on it first. If it is the case where we need this state to live around that's fine. This is really meant to be a defense in depth approach.